examples/check/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"time"
+)
+
+// this program is meant to be used to check the whole 'examples' cluster. It's meant to be put in the CI
+// so that new developments do not break the examples. We can see this as a single E2E test check
+// the program is supposed to start a transaction that flows through all the services under the 'examples'
+// folder and than check that the transaction actually triggered all the services in the chain until the
+// 'leaves'
+func main() {
+	// create simple client with timeout set to 1 second
+	client := http.Client{
+		Timeout: 1 * time.Second,
+	}
+
+	// perform a client call that is supposed to trigger a call across all the services
+	var jsonStr = []byte(`{"firstname":"John","lastname":"Doe"}`)
+	req, err := http.NewRequest("POST", "http://localhost:50000/api", bytes.NewBuffer(jsonStr))
+	if err != nil {
+		log.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if resp.StatusCode != http.StatusCreated {
+		log.Fatalf("Response supposed to be status code %d instead it is %d", http.StatusCreated, resp.StatusCode)
+	}
+
+	// wait 1 second so that metrics can be flushed to http endpoint
+	time.Sleep(1 * time.Second)
+
+	// services to be checked
+	svcsToUrls := map[string]string{
+		"http-svc":       "http://localhost:50000/",
+		"http-sec-svc":   "http://localhost:50001/",
+		"http-cache-svc": "http://localhost:50007/",
+		"kafka-svc":      "http://localhost:50002/",
+		"amqp-svc":       "http://localhost:50003/",
+		"sqs-svc":        "http://localhost:50004/",
+		"grpc-svc":       "http://localhost:50005/",
+	}
+
+	// collect the metrics for each of the services
+	// and early terminate in case of error
+	responses := make(map[string]*http.Response)
+	for svc, url := range svcsToUrls {
+		urlMetrics := url + "metrics"
+		log.Printf("Checking service %s [url %s]", svc, urlMetrics)
+		rsp, err := client.Get(urlMetrics)
+		if err != nil {
+			log.Fatalf("Svc %s is not reachable. HTTP error: ", err)
+		}
+		if rsp == nil {
+			log.Fatalf("Svc %s [url %s] lead to no response", svc, urlMetrics)
+		}
+		if rsp.StatusCode != http.StatusOK {
+			log.Fatalf("Svc %s [url %s] lead to HTTP status code : %d", svc, urlMetrics, rsp.StatusCode)
+		}
+		responses[svc] = rsp
+	}
+
+	// To check that the message flowed across all of the services we check the leaves of the message flow.
```

Comment:
I have mixed feelings about the way I'm making the check about the transaction reaching the leaf-services. I thought it might be at least a start for the check.

If you have more interesting suggestions, I welcome them.